app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package agency.five.codebase.android.movieapp.ui.moviedetails
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class MovieDetailsViewModel(
+    private val movieRepository: MovieRepository,
+    private val movieDetailsMapper: MovieDetailsMapper,
+    val movieId: Int
+) : ViewModel() {
+    private val _movieDetailsViewState = MutableStateFlow(MovieDetailsViewState.EMPTY)
+    val movieDetailsViewState: StateFlow<MovieDetailsViewState> =
+        _movieDetailsViewState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            movieRepository.movieDetails(movieId).collect {
+                _movieDetailsViewState.value = movieDetailsMapper.toMovieDetailsViewState(it)
+            }
+        }
+    }
```

Comment:
same as in HomeViewModel, remove init block and move it to variable initialization.